parser.go

```diff
@@ -234,21 +234,32 @@ func (parser *Parser) parseTypeSpec(pkgName string, typeSpec *ast.TypeSpec, prop
 			if field.Names == nil { //anonymous field
 				parser.parseAnonymousField(pkgName, field, properties)
 			} else {
-				name, propName := parser.parseField(field)
+				name, schemaType, arrayType := parser.parseField(field)
 				// if defined -- ref it
-				if _, ok := parser.TypeDefinitions[pkgName][propName]; ok {
+				if _, ok := parser.TypeDefinitions[pkgName][schemaType]; ok {
 					properties[name] = spec.Schema{
 						SchemaProps:
-						spec.SchemaProps{Type: []string{propName},
+						spec.SchemaProps{Type: []string{schemaType},
 							Ref: spec.Ref{
-								Ref: jsonreference.MustCreateRef("#/definitions/" + pkgName + "." + propName),
+								Ref: jsonreference.MustCreateRef("#/definitions/" + pkgName + "." + schemaType),
 							},
 						},
 					}
+				} else if schemaType == "array" {
+					// if defined -- ref it
+					if _, ok := parser.TypeDefinitions[pkgName][arrayType]; ok {
+						parser.ParseDefinition(pkgName, parser.TypeDefinitions[pkgName][arrayType] , arrayType)
+						properties[name] = spec.Schema{
+							SchemaProps: spec.SchemaProps{
+								Type: []string{schemaType},
+								Items: &spec.SchemaOrArray{Schema: &spec.Schema{SchemaProps: spec.SchemaProps{Ref: spec.Ref{ Ref: jsonreference.MustCreateRef("#/definitions/" + pkgName + "." + arrayType)}}}},
+							},
+						}
+					}
```

Comment:
248-256 
Seems like don't need reference it when schemaType == "array"

Reply:
User defined types should be available in parser.TypeDefinitions, as they need to be parsed and outputed:
`parser.ParseDefinition(pkgName, parser.TypeDefinitions[pkgName][arrayType] , arrayType)`

Actually there was still a missing case, the array of std. types (e.g. []string), which would be handled:
with an additional else:
```
} else { // standard type in array
						properties[name] = spec.Schema{
							SchemaProps:
							spec.SchemaProps{Type: []string{schemaType},
								Items: &spec.SchemaOrArray{Schema: &spec.Schema{SchemaProps: spec.SchemaProps{Type: []string{arrayType}}}}},
						}
					}
```

Resulting in:
```json
"Something": {
                    "type": "array",
                    "items": {
                        "type": "string"
                    }
                },
                "Templates": {
                    "type": "array",
                    "items": {
                        "$ref": "#/definitions/models.TagTemplate"
                    }
                }
```

complete method:
```
func (parser *Parser) parseTypeSpec(pkgName string, typeSpec *ast.TypeSpec, properties map[string]spec.Schema) {
	switch typeSpec.Type.(type) {
	case *ast.StructType:
		structDecl := typeSpec.Type.(*ast.StructType)
		fields := structDecl.Fields.List

		for _, field := range fields {
			if field.Names == nil { //anonymous field
				parser.parseAnonymousField(pkgName, field, properties)
			} else {
				name, schemaType, arrayType := parser.parseField(field)
				// TODO: find package of schemaType and/or arrayType

				if _, ok := parser.TypeDefinitions[pkgName][schemaType]; ok { // user type field
					properties[name] = spec.Schema{
						SchemaProps:
						spec.SchemaProps{Type: []string{schemaType},
							Ref: spec.Ref{
								Ref: jsonreference.MustCreateRef("#/definitions/" + pkgName + "." + schemaType),
							},
						},
					}
				} else if schemaType == "array" { // array field type
					// if defined -- ref it
					if _, ok := parser.TypeDefinitions[pkgName][arrayType]; ok { // user type in array
						parser.ParseDefinition(pkgName, parser.TypeDefinitions[pkgName][arrayType] , arrayType)
						properties[name] = spec.Schema{
							SchemaProps: spec.SchemaProps{
								Type: []string{schemaType},
								Items: &spec.SchemaOrArray{Schema: &spec.Schema{SchemaProps: spec.SchemaProps{Ref: spec.Ref{ Ref: jsonreference.MustCreateRef("#/definitions/" + pkgName + "." + arrayType)}}}},
							},
						}
					} else { // standard type in array
						properties[name] = spec.Schema{
							SchemaProps:
							spec.SchemaProps{Type: []string{schemaType},
								Items: &spec.SchemaOrArray{Schema: &spec.Schema{SchemaProps: spec.SchemaProps{Type: []string{arrayType}}}}},
						}
					}
				} else {
					// standard field type
					properties[name] = spec.Schema{
						SchemaProps:
						spec.SchemaProps{Type: []string{schemaType}},
					}
				}
			}
		}

	case *ast.ArrayType:
		log.Panic("ParseDefinitions not supported 'Array' yet.")
	case *ast.InterfaceType:
		log.Panic("ParseDefinitions not supported 'Interface' yet.")
	case *ast.MapType:
		log.Panic("ParseDefinitions not supported 'Map' yet.")
	}
}
```

Reply:
@nadilas  I appreciate your detail explanation. LGTM